src/lib/optimizer/join_ordering/enumerate_ccp.cpp

```diff
@@ -36,44 +38,42 @@ EnumerateCcp::EnumerateCcp(const size_t num_vertices, std::vector<std::pair<size
 
 std::vector<std::pair<JoinGraphVertexSet, JoinGraphVertexSet>> EnumerateCcp::operator()() {
   /**
-   * Initialize vertex neighborhood lookup table by computing and storing the neighborhood of each vertex
+   * Initialize vertex neighborhood lookup table by computing and storing the neighborhood of each vertex.
    */
   _vertex_neighborhoods.resize(_num_vertices);
   for (auto vertex_idx = size_t{0}; vertex_idx < _num_vertices; ++vertex_idx) {
     _vertex_neighborhoods[vertex_idx] = _single_vertex_neighborhood(vertex_idx);
   }
 
   /**
-   * This loop corresponds to EnumerateCsg in the paper
+   * This loop corresponds to EnumerateCsg in the paper.
    *
-   * It iterate from the highest to the lowest vertex index and starts a search for connected subgraphs from
-   * each vertex (_enumerate_csg_recursive()).
-   * For each subgraph, a search for complement subgraphs is started (_enumerate_cmp()).
+   * It iterates from the highest to the lowest vertex index and starts a search for connected subgraphs from
+   * each vertex (`_enumerate_csg_recursive()`). For each subgraph, a search for complement subgraphs is started
+   * (`_enumerate_cmp()`).
    */
-  for (auto reverse_vertex_idx = size_t{0}; reverse_vertex_idx < _num_vertices; ++reverse_vertex_idx) {
-    const auto forward_vertex_idx = _num_vertices - reverse_vertex_idx - 1;
-
+  for (auto vertex_idx = _num_vertices; vertex_idx > 0; --vertex_idx) {
```

Comment:
From a readability perspective I would start the loop with `_num_vertices - 1`, stop at `>= 0`, and simply use `vertex_idx` instead of `vertex_idx - 1`.